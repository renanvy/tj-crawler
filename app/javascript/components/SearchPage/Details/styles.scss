.Details {
  &_title {
    margin: 0;
    margin-top: 5px;
    font-size: 18px;
    font-weight: 500;
  }

  &_list_infos li {
    margin-bottom: 10px;
  }

  &_involved_list li {
    margin-bottom: 10px;
  }
}