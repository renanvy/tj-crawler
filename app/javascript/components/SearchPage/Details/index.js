import React from "react";
import PropTypes from "prop-types";

import "./styles.scss";

export default function Details(props) {
  const { process } = props;
  return (
    <div className="col s12 m4">
      <h1 className="Details_title">Detalhes do processo</h1>

      <ul className="Details_list_infos">
        <li>
          <strong>Distribuído em:</strong>
          {process.distribution_date}
        </li>
        <li>
          <strong>Classe: </strong>
          {process.kind}
        </li>
        <li>
          <strong>Área: </strong>
          {process.area}
        </li>
        <li>
          <strong>Assunto: </strong>
          {process.subject}
        </li>
        <li>
          <strong>Juíz: </strong>
          {process.judge}
        </li>
        <li>
          <strong>Valor da Ação: </strong>
          {process.action_value}
        </li>
      </ul>

      <div className="divider" />

      <br />

      <h1 className="Details_title">Partes envolvida</h1>

      <ul className="Details_involved_list">
        <li>{process.plaintiff}</li>
        <li>{process.sued}</li>
      </ul>
    </div>
  );
}

Details.propTypes = {
  process: PropTypes.object
};
