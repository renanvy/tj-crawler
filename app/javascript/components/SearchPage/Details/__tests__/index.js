import React from "react";
import renderer from "react-test-renderer";
import Details from "../index";

describe("Details", () => {
  test("snapshot renders", () => {
    const process = { number: "1234" };

    const component = renderer.create(<Details process={process} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
