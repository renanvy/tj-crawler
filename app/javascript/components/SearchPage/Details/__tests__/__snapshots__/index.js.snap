// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details snapshot renders 1`] = `
<div
  className="col s12 m4"
>
  <h1
    className="Details_title"
  >
    Detalhes do processo
  </h1>
  <ul
    className="Details_list_infos"
  >
    <li>
      <strong>
        Distribuído em:
      </strong>
    </li>
    <li>
      <strong>
        Classe: 
      </strong>
    </li>
    <li>
      <strong>
        Área: 
      </strong>
    </li>
    <li>
      <strong>
        Assunto: 
      </strong>
    </li>
    <li>
      <strong>
        Juíz: 
      </strong>
    </li>
    <li>
      <strong>
        Valor da Ação: 
      </strong>
    </li>
  </ul>
  <div
    className="divider"
  />
  <br />
  <h1
    className="Details_title"
  >
    Partes envolvida
  </h1>
  <ul
    className="Details_involved_list"
  >
    <li />
    <li />
  </ul>
</div>
`;
